pallets/kensetsu/src/lib.rs

```diff
@@ -1149,49 +1140,76 @@ pub mod pallet {
-                        Error::<T>::AccrueWrongTime
-                    );
-                    // do not update if time is the same
-                    if now > collateral_info.last_fee_update_time {
-                        let time_passed = now
-                            .checked_sub(&collateral_info.last_fee_update_time)
-                            .ok_or(Error::<T>::ArithmeticError)?;
-                        let new_coefficient = compound(
-                            collateral_info.interest_coefficient.into_inner(),
-                            collateral_info.risk_parameters.stability_fee_rate,
-                            time_passed
-                                .checked_into::<u64>()
-                                .ok_or(Error::<T>::ArithmeticError)?,
-                        )
-                        .map_err(|_| Error::<T>::ArithmeticError)?;
-                        collateral_info.last_fee_update_time = now;
-                        collateral_info.interest_coefficient =
-                            FixedU128::from_inner(new_coefficient);
-                    }
-                    Ok::<CollateralInfo<T::Moment>, DispatchError>(collateral_info.clone())
-                })?;
+            let mut collateral_info = CollateralInfos::<T>::get(collateral_asset_id)
+                .ok_or(Error::<T>::CollateralInfoNotFound)?;
+            let now = Timestamp::<T>::get();
+            ensure!(
+                now >= collateral_info.last_fee_update_time,
+                Error::<T>::AccrueWrongTime
+            );
 
+            // do not update if time is the same
+            if now > collateral_info.last_fee_update_time {
+                let time_passed = now
+                    .checked_sub(&collateral_info.last_fee_update_time)
+                    .ok_or(Error::<T>::ArithmeticError)?;
+                let new_coefficient = compound(
+                    collateral_info.interest_coefficient.into_inner(),
+                    collateral_info.risk_parameters.stability_fee_rate,
+                    time_passed
+                        .checked_into::<u64>()
+                        .ok_or(Error::<T>::ArithmeticError)?,
+                )
+                .map_err(|_| Error::<T>::ArithmeticError)?;
+                collateral_info.last_fee_update_time = now;
+                collateral_info.interest_coefficient = FixedU128::from_inner(new_coefficient);
+            }
             Ok(collateral_info)
         }
 
+        /// Recalculates and updates collateral interest coefficient with the current timestamp
+        fn update_collateral_interest_coefficient(
+            collateral_asset_id: &AssetIdOf<T>,
+        ) -> Result<CollateralInfo<T::Moment>, DispatchError> {
+            let updated_collateral_info =
+                Self::calculate_collateral_interest_coefficient(collateral_asset_id)?;
+            <CollateralInfos<T>>::set(collateral_asset_id, Some(updated_collateral_info.clone()));
```

Comment:
It is called even if  `collateral_info` was not changed.
I don't know, probably Substrate checks it and doesn't write the same value, but if not - there is the unnecessary `write`.

Reply:
Yes, good point. This write may be removed at all and calculated in runtime.